agent/engine/execcmd/manager_init_task_linux.go

```diff
@@ -284,8 +284,7 @@ func getAgentConfigFileName(sessionLimit int) (string, error) {
 	configFileName := fmt.Sprintf(execAgentConfigFileNameTemplate, hash)
 	// check if config file exists already
 	configFilePath := filepath.Join(ECSAgentExecConfigDir, configFileName)
-	if fileExists(configFilePath) {
-		// TODO: verify the hash of the existing file contents
+	if fileExists(configFilePath) && validConfigExists(configFilePath, hash) {
```

Comment:
maybe this is for a new PR, but can you check what happens when a directory with the same name as the config file already exists on disk?

My guess is that `createNewExecAgentConfigFile` will fail. I think maybe we need to make sure we remove a directory (or an existing file which is tampered with) before writing the new file

Reply:
today we return an error for the following:
- if the file is a directory
- if the file is tampered 

Are you suggesting we do not return an error in these cases?

Reply:
will change to delete the directory